status/term.go

```diff
@@ -133,7 +133,7 @@ type winsize struct {
 
 func getWinsize(fd uintptr) (*winsize, error) {
 	w := new(winsize)
-	_, _, err := unix.Syscall(unix.SYS_IOCTL, fd, uintptr(unix.TIOCGWINSZ), uintptr(unsafe.Pointer(w)))
+	_, _, err := unix.Syscall(syscall.SYS_IOCTL, fd, uintptr(unix.TIOCGWINSZ), uintptr(unsafe.Pointer(w)))
```

Comment:
```suggestion
	_, _, err := unix.Syscall(unix.SYS_IOCTL, fd, uintptr(unix.TIOCGWINSZ), uintptr(unsafe.Pointer(w)))
```

Reply:
My understanding is that `unix` should be favored over the no-longer-updated `syscall`.

Reply:
This symbol couldn't be found after upgrading `golang.org/x/sys`. It still appears to be defined in `golang.org/x/sys/unix` on master so I'm not entirely sure why it's causing compilation issues. I can try to figure out an alternate solution if you'd like to avoid the `syscall` package.

Reply:
I think it would be good to understand what's happening here. The documentation for the specific version in `go.mod` has `SYS_IOCTL`: https://pkg.go.dev/golang.org/x/sys@v0.0.0-20200923182605-d9f96fdee20d/unix .

Reply:
It looks like `SYS_IOCTL` isn't defined in any of the `unix/zsyscall_*.go` files which are included on darwin builds. I can try to go back through the history of `golang.org/x/sys` to figure out when they were removed but this used to compile in our mac build environment.

Reply:
It looks like darwin support for these syscall constants was removed in https://github.com/golang/sys/commit/6fcdbc0bbc04dcd9e7dc145879ceaf9bf1c6ff03#diff-4913a9178621eadcdf191db17915fbcb causing the build errors.

Reply:
I see. I looked into it a bit, and I think this call should eventually be replaced with a call to `unix.IoctlGetWinsize`. I have written more detail in https://github.com/grailbio/base/issues/26. Does just changing it to use `syscall.SYS_IOCTL` actually work (beyond compiling)? I'm not very well-versed in Go syscalls, so it's not obvious to me.

Reply:
`syscall.SYS_IOCTL` is just an integer constant with value 54. `unix.SYS_IOCTL` had the same value (54) in the version of golang.org/x/sys/unix which was in use before this PR. I'm also not super familiar with syscalls in go but I believe it should be safe.

Reply:
Yes; I understand that it's the same value that it was. However, if I understand correctly, it was removed because direct syscalls are no longer supported on Darwin, which could either mean that 1. it still works but is just not officially supported 2. it still "works" but there is some observable consequence, e.g. can't be distributed in the app store or 3. it actually doesn't work any more.

I'm not familiar enough with the unit tests to know whether this is exercised, so I think we should either verify that the unit tests are sufficient or manually test. I can get to this early next week if you don't want to dig into it. I know that people use `base/status` on Darwin, so I want to make sure that we don't break it.

Reply:
Ah, that makes sense. I'm not at all familiar with `base/status` but a quick scan of the tests in `status_test.go` make it seem like this code path isn't exercised. I can try to dig into `base/status` and figure out how to exercise this code if I have time.

Reply:
Great; thanks, and sorry that it's taking so long! I'll ping back here if I'm able to take a look Monday morning. Also, to be clear, I'm only worried about (3) on 1.13. It also seems like porting to `unix.IoctlGetWinsize` might be very straightforward, once we have a way to test it.

Reply:
Hey @jcharum, have you had a chance to look into testing `base/status`?

Reply:
I just made a PR on this branch with a fix-forward to use `IoctlGetWinsize`. I tested it with [this simple program](https://gist.github.com/jcharum/0a5a07122df320570af49a2c61fa355b).

Reply:
Thank you!

Reply:
I've merged the PR